[Bug] java.lang.NumberFormatException: For input string: "30s"

when submitted the flink jar application，it run failed, the error log as below:

_environments info： flink-1.12.2，CDH-6.3.2（hadoop ），java-1.8，scala-2.11.12_

> java.lang.reflect.InvocationTargetException
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at com.streamxhub.streamx.flink.submit.FlinkSubmitHelper$$anonfun$submit$1.apply(FlinkSubmitHelper.scala:51)
> 	at com.streamxhub.streamx.flink.submit.FlinkSubmitHelper$$anonfun$submit$1.apply(FlinkSubmitHelper.scala:46)
> 	at com.streamxhub.streamx.flink.proxy.FlinkShimsProxy$$anonfun$proxy$1.apply(FlinkShimsProxy.scala:68)
> 	at com.streamxhub.streamx.common.util.ClassLoaderUtils$.runAsClassLoader(ClassLoaderUtils.scala:42)
> 	at com.streamxhub.streamx.flink.proxy.FlinkShimsProxy$.proxy(FlinkShimsProxy.scala:68)
> 	at com.streamxhub.streamx.flink.submit.FlinkSubmitHelper$.submit(FlinkSubmitHelper.scala:46)
> 	at com.streamxhub.streamx.flink.submit.FlinkSubmitHelper.submit(FlinkSubmitHelper.scala)
> 	at com.streamxhub.streamx.console.core.service.impl.ApplicationServiceImpl.start(ApplicationServiceImpl.java:1187)
> 	at com.streamxhub.streamx.console.core.service.impl.ApplicationServiceImpl$$FastClassBySpringCGLIB$$4cca1efc.invoke(<generated>)
> 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:88)
> 	at com.streamxhub.streamx.console.core.aspect.StreamXConsoleAspect.lambda$refreshCache$0(StreamXConsoleAspect.java:91)
> 	at com.streamxhub.streamx.console.core.task.FlinkTrackingTask.refreshTracking(FlinkTrackingTask.java:629)
> 	at com.streamxhub.streamx.console.core.aspect.StreamXConsoleAspect.refreshCache(StreamXConsoleAspect.java:89)
> 	at sun.reflect.GeneratedMethodAccessor297.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:644)
> 	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:633)
> 	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:367)
> 	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:118)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691)
> 	at com.streamxhub.streamx.console.core.service.impl.ApplicationServiceImpl$$EnhancerBySpringCGLIB$$1e8e9325.start(<generated>)
> 	at com.streamxhub.streamx.console.core.controller.ApplicationController.start(ApplicationController.java:139)
> 	at com.streamxhub.streamx.console.core.controller.ApplicationController$$FastClassBySpringCGLIB$$a9cf358.invoke(<generated>)
> 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.validation.beanvalidation.MethodValidationInterceptor.invoke(MethodValidationInterceptor.java:119)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:88)
> 	at com.streamxhub.streamx.console.core.aspect.StreamXConsoleAspect.response(StreamXConsoleAspect.java:64)
> 	at sun.reflect.GeneratedMethodAccessor114.invoke(Unknown Source)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:644)
> 	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:633)
> 	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.apache.shiro.spring.security.interceptor.AopAllianceAnnotationsAuthorizingMethodInterceptor$1.proceed(AopAllianceAnnotationsAuthorizingMethodInterceptor.java:82)
> 	at org.apache.shiro.authz.aop.AuthorizingMethodInterceptor.invoke(AuthorizingMethodInterceptor.java:39)
> 	at org.apache.shiro.spring.security.interceptor.AopAllianceAnnotationsAuthorizingMethodInterceptor.invoke(AopAllianceAnnotationsAuthorizingMethodInterceptor.java:115)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95)
> 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
> 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
> 	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691)
> 	at com.streamxhub.streamx.console.core.controller.ApplicationController$$EnhancerBySpringCGLIB$$a67b7ab.start(<generated>)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
> 	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
> 	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
> 	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878)
> 	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792)
> 	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
> 	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
> 	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
> 	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
> 	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:517)
> 	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
> 	at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
> 	at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
> 	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
> 	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:61)
> 	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108)
> 	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137)
> 	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
> 	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66)
> 	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108)
> 	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137)
> 	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
> 	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66)
> 	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:450)
> 	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
> 	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
> 	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
> 	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:387)
> 	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
> 	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
> 	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
> 	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
> 	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:92)
> 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
> 	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
> 	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
> 	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
> 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
> 	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
> 	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
> 	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
> 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
> 	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
> 	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
> 	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
> 	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
> 	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
> 	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
> 	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
> 	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
> 	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
> 	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
> 	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
> 	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:111)
> 	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
> 	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
> 	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
> 	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
> 	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
> 	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
> 	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
> 	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
> 	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
> 	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:269)
> 	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:78)
> 	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:133)
> 	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:130)
> 	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
> 	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
> 	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:249)
> 	at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:78)
> 	at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:99)
> 	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:370)
> 	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:836)
> 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
> 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2019)
> 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1558)
> 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1449)
> 	at java.lang.Thread.run(Thread.java:748)
> Caused by: java.lang.reflect.UndeclaredThrowableException
> 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1944)
> 	at org.apache.flink.runtime.security.contexts.HadoopSecurityContext.runSecured(HadoopSecurityContext.java:41)
> 	at com.streamxhub.streamx.flink.submit.impl.YarnApplicationSubmit$.doSubmit(YarnApplicationSubmit.scala:64)
> 	at com.streamxhub.streamx.flink.submit.trait.FlinkSubmitTrait$class.submit(FlinkSubmitTrait.scala:74)
> 	at com.streamxhub.streamx.flink.submit.impl.YarnApplicationSubmit$.submit(YarnApplicationSubmit.scala:48)
> 	at com.streamxhub.streamx.flink.submit.FlinkSubmit$.submit(FlinkSubmit.scala:31)
> 	at com.streamxhub.streamx.flink.submit.FlinkSubmit.submit(FlinkSubmit.scala)
> 	... 151 more
> Caused by: org.apache.flink.client.deployment.ClusterDeploymentException: Couldn't deploy Yarn Application Cluster
> 	at org.apache.flink.yarn.YarnClusterDescriptor.deployApplicationCluster(YarnClusterDescriptor.java:465)
> 	at com.streamxhub.streamx.flink.submit.impl.YarnApplicationSubmit$$anon$1.call(YarnApplicationSubmit.scala:80)
> 	at com.streamxhub.streamx.flink.submit.impl.YarnApplicationSubmit$$anon$1.call(YarnApplicationSubmit.scala:64)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at javax.security.auth.Subject.doAs(Subject.java:422)
> 	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1926)
> 	... 157 more
> Caused by: java.lang.NumberFormatException: For input string: "30s"
> 	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
> 	at java.lang.Long.parseLong(Long.java:589)
> 	at java.lang.Long.parseLong(Long.java:631)
> 	at org.apache.hadoop.conf.Configuration.getLong(Configuration.java:1435)
> 	at org.apache.hadoop.hdfs.client.impl.DfsClientConf.<init>(DfsClientConf.java:255)
> 	at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:319)
> 	at org.apache.hadoop.hdfs.DFSClient.<init>(DFSClient.java:303)
> 	at org.apache.hadoop.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:159)
> 	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3247)
> 	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:121)
> 	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3296)
> 	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3264)
> 	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:475)
> 	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:228)
> 	at org.apache.flink.yarn.YarnClusterDescriptor.startAppMaster(YarnClusterDescriptor.java:769)
> 	at org.apache.flink.yarn.YarnClusterDescriptor.deployInternal(YarnClusterDescriptor.java:592)
> 	at org.apache.flink.yarn.YarnClusterDescriptor.deployApplicationCluster(YarnClusterDescriptor.java:458)
> 	... 162 more

> **1.hadoop版本： 3.0.0**
> 
> **2.hadoop配置文件** [core-site.txt]( [https://github ](https://github) [hdfs-site.txt](https://github.com/streamxhub/streamx/files/7738759/hdfs-site.txt) .com/streamxhub/streamx/files/7738758/core-site.txt)
> 
> **3.该长值的参数应该配到哪里呢** CDH的管理界面CM上hdfs所有时间相关的配置默认都带了时间单位，改不了的

请问您这个问题最后解决了嘛?  如果解决了, 能分享一下是如何解决的嘛, 感谢! 

 我的环境也是CDH6.3.2, hdsf-site与core-site文件里和时间有关的属性我也都修改了(我是在CM界面里的高级配置代码段修改的).

具体属性修改见下方链接:  https://github.com/apache/hadoop/commit/3f573b6751a726c0a30de3f62a0984bd43b15a53

修改完且重启集群后, 启动streamx SQL任务还是报错(我的JAR任务可以正常运行) :  
Caused by: java.lang.IllegalArgumentException: [StreamX] access hdfs error: java.lang.NumberFormatException: For input string: "30s".

--- a/streampark-common/src/main/scala/org/apache/streampark/common/util/HadoopUtils.scala
+++ b/streampark-common/src/main/scala/org/apache/streampark/common/util/HadoopUtils.scala
@@ -125,29 +125,9 @@ object HadoopUtils extends Logger {
       val hadoopConfDir = new File(confDir)
       val confName = List("hdfs-default.xml", "core-site.xml", "hdfs-site.xml", "yarn-site.xml")
       val files = hadoopConfDir.listFiles().filter(x => x.isFile && confName.contains(x.getName)).toList
-      val conf = new Configuration()
+      val conf = new HadoopConfiguration()
       if (CollectionUtils.isNotEmpty(files)) {
         files.foreach(x => conf.addResource(new Path(x.getAbsolutePath)))
-        // HDFS default value change (with adding time unit) breaks old version MR tarball work with Hadoop 3.x
-        // detail: https://issues.apache.org/jira/browse/HDFS-12920
-        val rewriteNames = List(
-          "dfs.blockreport.initialDelay",
-          "dfs.datanode.directoryscan.interval",
-          "dfs.heartbeat.interval",
-          "dfs.namenode.decommission.interval",
-          "dfs.namenode.replication.interval",
-          "dfs.namenode.checkpoint.period",
-          "dfs.namenode.checkpoint.check.period",
-          "dfs.client.datanode-restart.timeout",
-          "dfs.ha.log-roll.period",
-          "dfs.ha.tail-edits.period",
-          "dfs.datanode.bp-ready.timeout")
-        rewriteNames.foreach(n => {
-          Option(conf.get(n)) match {
-            case Some(v) if v.matches("\\d+s$") => conf.set(n, v.dropRight(1))
-            case _ =>
-          }
-        })
       }
       configurationCache.put(confDir, conf)
     }
@@ -289,5 +269,60 @@ object HadoopUtils extends Logger {
     val destPath = new Path(tmpDir.getAbsolutePath + "/" + sourcePath.getName)
     fs.copyToLocalFile(sourcePath, destPath)
     new File(destPath.toString).getAbsolutePath
+  }
+
+  private class HadoopConfiguration extends Configuration {
+
+    private lazy val rewriteNames = List(
+      "dfs.blockreport.initialDelay",
+      "dfs.datanode.directoryscan.interval",
+      "dfs.heartbeat.interval",
+      "dfs.namenode.decommission.interval",
+      "dfs.namenode.replication.interval",
+      "dfs.namenode.checkpoint.period",
+      "dfs.namenode.checkpoint.check.period",
+      "dfs.client.datanode-restart.timeout",
+      "dfs.ha.log-roll.period",
+      "dfs.ha.tail-edits.period",
+      "dfs.datanode.bp-ready.timeout")
+
+    private def getHexDigits(value: String): String = {
+      var negative = false
+      var str = value
+      var hexString: String = null
+      if (value.startsWith("-")) {
+        negative = true
+        str = value.substring(1)
+      }
+      if (str.startsWith("0x") || str.startsWith("0X")) {
+        hexString = str.substring(2)
+        if (negative) hexString = "-" + hexString
+        return hexString
+      }
+      null
+    }
+
+    // HDFS default value change (with adding time unit) breaks old version MR tarball work with Hadoop 3.x
+    // detail: https://issues.apache.org/jira/browse/HDFS-12920
+    private def getSafeValue(name: String): String = {
+      val value = getTrimmed(name)
+      if (rewriteNames.contains(name) && name.matches("\\d+s$")) {
+        return value.dropRight(1)
+      }
+      value
+    }
+
+    override def getLong(name: String, defaultValue: Long): Long = {
+      val valueString = getSafeValue(name)
+      valueString match {
+        case null => defaultValue
+        case v =>
+          val hexString = getHexDigits(v)
+          if (hexString != null) {
+            return java.lang.Long.parseLong(hexString, 16)
+          }
+          java.lang.Long.parseLong(valueString)
+      }
+    }
   }
 }

--- a/streampark-common/src/main/scala/org/apache/streampark/common/util/HadoopUtils.scala
+++ b/streampark-common/src/main/scala/org/apache/streampark/common/util/HadoopUtils.scala
@@ -307,7 +307,7 @@ object HadoopUtils extends Logger {
     private def getSafeValue(name: String): String = {
       val value = getTrimmed(name)
       if (rewriteNames.contains(name)) {
-        return value.replaceFirst("\\s+s$", "")
+        return value.replaceFirst("s$", "")
       }
       value
     }

--- a/streampark-common/src/main/scala/org/apache/streampark/common/util/HadoopUtils.scala
+++ b/streampark-common/src/main/scala/org/apache/streampark/common/util/HadoopUtils.scala
@@ -306,8 +306,8 @@ object HadoopUtils extends Logger {
     // detail: https://issues.apache.org/jira/browse/HDFS-12920
     private def getSafeValue(name: String): String = {
       val value = getTrimmed(name)
-      if (rewriteNames.contains(name) && name.matches("\\d+s$")) {
-        return value.dropRight(1)
+      if (rewriteNames.contains(name)) {
+        return value.replaceFirst("\\s+s$", "")
       }
       value
     }

